vllm/forward_context.py

```diff
@@ -66,4 +56,37 @@ def set_forward_context(context: Any, vllm_config: VllmConfig):
     try:
         yield
     finally:
+        global batchsize_counter
+        global last_logging_time, batchsize_logging_interval
+        if need_to_track_batchsize:
+            if hasattr(context, "num_prefill_tokens"):
+                # for v0 attention backends
+                batchsize = context.num_prefill_tokens + \
+                    context.num_decode_tokens
+            else:
+                # for v1 attention backends
+                batchsize = context.num_input_tokens
+            # we use synchronous scheduling right now,
+            # adding a sync point here should not affect
+            # scheduling of the next batch
+            torch.cuda.synchronize()
```

Comment:
I need to look at how this function is called to agree with this usage.. do you know what is the right thing to do in V1 as we won't have synchronous?

Reply:
> do you know what is the right thing to do in V1 as we won't have synchronous?

can you elaborate that?

> I need to look at how this function is called to agree with this usage..

this context manager `set_forward_context` is used for every model forward, and later we will pull the result from GPU to CPU.

Reply:
My point is this will cause unrealistic behavior if we are running with the asynchronous scheduler, but maybe it is not a big deal. Measuring performance will always have a tradeoff from realism but this is likely the best we can do

Reply:
that's true. perf measurement itself always has overhead. as long as it does not hurt the case when perf measurement is disabled, then it is fine imo.